app/controllers/users_controller.rb

```diff
@@ -16,7 +18,30 @@ def update
     end
   end
 
+  def show_camps
+    @camps = Camp.where(status: "active")
+  end
+
+  def select_camp
+    if CampUser.find_by(user_id: params[:id])
+      flash[:notice] = "You have already selected camp"
+      redirect_to user_personal_info_path
+    else
+      @camp_user = CampUser.new(camp_users_params)
+      @camp_user.user_id = params[:id]
+      @camp_user.save
+      flash[:notice] = "Camp Selected"
+      redirect_to user_personal_path
+    end
+  end
+
```

Comment:
Please consider updating it like follow: 

def select_camp
  return redirect_to user_personal_info_path, notice: "You have already selected camp" if CampUser.find_by(user_id: params[:id]).present?

  @camp_user = CampUser.new(camp_users_params)
  @camp_user.user_id = params[:id]
  if @camp_user.save
    redirect_to user_personal_path, notice: "Camp Selected"
  else
    redirect_to user_personal_path, alert: @camp_user.errors_.full_messages.to_sentence
  end
end